projects/fal/src/faldbt/lib.py

```diff
@@ -84,27 +105,13 @@ def _get_adapter(
     return adapter
 
 
-global _lock
-# RLock supports recursive locking by the same thread
-_lock = threading.RLock()
-
-
 @contextmanager
-def _cache_lock(info: str = ""):
+def _cache_lock(adapter, info: str = ""):
     operationId = uuid4()
     LOGGER.debug("Locking  {} {}", operationId, info)
 
-    _lock.acquire()
-    LOGGER.debug("Acquired {}", operationId)
-
-    try:
+    with adapter.cache.lock:
         yield
-    except:
-        LOGGER.debug("Error during lock operation {}", operationId)
-        raise
-    finally:
-        _lock.release()
-        LOGGER.debug("Released {}", operationId)
```

Comment:
Point of interest: this `RLock` when acquired in an isolated environment and with the `dbt-fal` adapter, it would not acquire it ever, instead it would hang in `_lock.acquire()` forever.

I am not sure what caused it and changed it to rely on `adapter.cache.lock` for now.

Any ideas on why this may have been? I don't love relying on the internal adapter lock.

Reply:
@isidentical found the reason! https://github.com/uqfoundation/dill/issues/321

Reply:
reverted by using `threding._PyRLock`